### ISSUE TITLE
No code assist on enum value while typing a switch statement

### ISSUE THREAD

**Issue by mauromol** [groovy/groovy-eclipse#762]
Consider the following enum:

```java
package test40;
public enum MyEnum {
	FOO, BAR;
}
```

And the following Groovy class:

```groovy
package test40
class Test40 {
	void foo() {
		MyEnum e = MyEnum.FOO
		switch(e) {
			case MyEnum.FO|
		}
	}
}
```

Let the cursor be at "|" and invoke code assist co complete `MyEnum.FOO`: no suggestion is given.

**Comment by eric-milles** [comment]
Ready to test.

Some things are not going to work as you may expect, like:
```groovy
switch (e) {
case |
}
```
will probably not propose the enum type.  And completion will not hide enum values that are already present in other case labels.  And fields, methods, etc. that match the switch type are not boosted to the top of the proposals.

Lastly, Groovy does not support the same form of switch for enums:
```java
switch (e) {
case ONE:
case TWO:
}
```
I have requested this support in the past; you can track progress if interested: https://issues.apache.org/jira/browse/GROOVY-8444

**Comment by mauromol** [comment]
The described use case is fixed in 3.2.0.xx-201811182243-e48, but indeed having code assist propose the enum values when no character is typed would be desirable. I'm going to open a new issue for that.
